interview/algorithm/sgd.py

```diff
@@ -0,0 +1,61 @@
+import numpy as np
+
+
+class SGD:
+    """Batch Stochastic Gradien Descent iterative optimization process. It is one of
+    the most used methods for identifying the parameters of the model that minimize the
+    cost function in machine learning algorithms.
+
+    :param float learning_rate: factor controlling the magnitude of the vector update.
+    :param float decay_rate: factor setting contribution of the previous update.
+    :param float max_iteration: number of iterations.
+    :param float batch_size: size of minibatches.
+    :param float tolerance: minimal allowed movement in each iteration.
+    :param int seed: seed to randomly select observations for minibatches.
+    """
+
+    def __init__(
+        self,
+        learning_rate=0.01,
+        decay_rate=0,
+        max_iteration=1000,
+        batch_size=1,
+        tolerance=1e-3,
+        seed=None,
+    ):
+        self.learning_rate = learning_rate
+        self.decay_rate = decay_rate
+        self.max_iteration = max_iteration
+        self.batch_size = batch_size
+        self.tolerance = tolerance
+        self.seed = None if seed is None else seed
```

Comment:
Why are you doing this? Is self.seed not given by the input seed?

Reply:
Not sure what I was trying to do here.

Reply:
Done